cmake/OpenDDSConfig.cmake

```diff
@@ -137,6 +137,14 @@ if(OPENDDS_XERCES3)
   endif()
 endif()
 
+## ACE provides (or uses) many deprecated functions.
+## For now, we'll silence the warnings.
+if(MSVC)
+  add_compile_options(/wd4996)
+  add_compile_definitions(_CRT_SECURE_NO_WARNINGS
+                          _WINSOCK_DEPRECATED_NO_WARNINGS)
```

Comment:
These should probably be attached to the ACE library if that's what they're meant for. This could be moved to the bottom after the ACE library is defined, probably at the end of the `if(OPENDDS_FOUND)` block and changed to something like:

```
target_compile_options(ACE::ACE INTERFACE /wd4996)
target_compile_definitions(ACE::ACE INTERFACE _CRT_SECURE_NO_WARNINGS _WINSOCK_DEPRECATED_NO_WARNINGS)
```

Reply:
They cannot be attached only to the ACE library.  Much, if not all, of what is warned about are in inline functions.  Otherwise, they wouldn't actually show up when building the cmake tests.

Reply:
I think I misunderstood what you were saying.  But, still, I'm not quite sure why it matters.

Reply:
It matters because it's generally preferred to use the target-specific functions and avoid global ones. [We have had to remove a global function for adding includes in the past](https://github.com/objectcomputing/OpenDDS/issues/1395). In this case even if ACE is going to be used by everything we use, the user the might have non-ACE targets and these flags shouldn't be applied to those.